keps/0034-instance-health.md

```diff
@@ -106,10 +104,16 @@ From all these resources it would compute readiness based on `Condition: Ready`,
 
 For operators installing their app via e.g. `Deployment`, there’s no need at this point to also check the Status of the underlying Pods because Deployment `Status` already mirrors the readiness of the underlying Pods. This is true to all higher-level types like StatefulSet etc.
 
+To mitigate situation when a plan adds a resource we don't want to include in `Ready` condition, it's possible to opt-out from this when adding annotation `kudo.dev/allow-readiness-computation:false`
+
 ### Risks and Mitigations
 
 On a big cluster with a huge number of Pods and Deployments, it’s possible that a controller might have a scaling issues because of number of items it need to process.
 
 This is mitigated by a fact that inside event handlers we’re filtering only for events that belong to KUDO which should limit the scope of events to only a very few (the expectation is that majority of the Deployments does not come from KUDO).
 
 Also this feature is not making this situation any worse as it’s right now because KUDO controller already watches for Pods and Deployments, so we’re not introducing new overhead. That said the controller will have to perform much more work in times where it was just "idling" before - because it was working only when plan was run, otherwise the reconcile ended right after it started. This could pose problem on bigger clusters with many KUDO operators and could be mitigated by running KUDO controller with multiple workers (right now 1 worker is enough on most installations we're aware of).
+
+### Future work
+
+To support even more complex cases and operator, we should think about adding possibility to specify `kudo.dev/readiness-strategy:selected` which will switch readiness computation to OPT-IN mode. With this one set, only resources with `kudo.dev/allow-readiness-computation:true` will be considered for readiness computation.
```

Comment:
Ok, having read this the annotation from above makes a little more sense, still... Maybe we can find something that works with better defaults? Maybe we need two different annotations for opt-in and opt-out?

Reply:
Alright I kept both in place (two different options).